File: 1-TwoSum/1-TwoSum.java
// Last updated: 7/29/2026, 10:14:15 AM
class Solution {
    public int[] twoSum(int[] nums, int target) {
        for(int i = 0 ; i < nums.length; i++){
            for(int j = 1; j < nums.length; j++){
                if( i != j && nums[i] + nums[j] == target){
                    int[] myArray = new int[2];
                    myArray[0] = i;
                    myArray[1] = j;
                    return myArray;
                }
            }
        }
        return nums;
    }
}

File: 1-TwoSum/1-TwoSum.cpp
// Last updated: 7/29/2026, 10:14:15 AM
#include <vector>
#include <unordered_map>
using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> numberMap;
        for (int i = 0; i < nums.size(); i++) {
           if(numberMap.find(target - nums[i]) != numberMap.end()){
            return {numberMap[target - nums[i]], i};
           }
           numberMap[nums[i]] = i;
        }
        return {};  // Return an empty vector if no solution is found
    }
};
